node/src/main/scala/org/bitcoins/node/PeerManager.scala

```diff
@@ -485,9 +486,20 @@ case class PeerManager(
         logger.debug(s"Got ${payload.commandName} from peer=${peer} in stream")
         getDataMessageHandler
           .handleDataPayload(payload, peerMsgSender, peer)
-          .map { newDmh =>
-            updateDataMessageHandler(newDmh)
-            msg
+          .flatMap { newDmh =>
+            newDmh.state match {
+              case m: MisbehavingPeer =>
+                updateDataMessageHandler(newDmh)
+                //disconnect the misbehaving peer
```

Comment:
Here is where we are now syncing with a new peer based on the return value from `DataMessageHandler.handleDataPayload()`